sojo-api/app/controllers/PaymentHistoryController.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+
+const blueprint          = require ('@onehilltech/blueprint')
+,     mongodb            = require ('@onehilltech/blueprint-mongodb')
+,     ResourceController = mongodb.ResourceController
+,     PaymentHistory     = require ('../models/PaymentHistory')
+;
+
+function PaymentHistoryController () {
+  ResourceController.call (this, { model: PaymentHistory });
+}
+
+PaymentHistoryController.prototype.__defineGetter__ ('resourceId', () => {
+  return 'Id';
+});
+
+PaymentHistoryController.prototype.getAll = (args) => {
+  return {
+    execute: (req, res, callback) => {
+      PaymentHistory.find ({account_id: req.params.Id }, (err, payment_history) => {
+        if (err) {
+          res.status (404).json (err);
```

Comment:
We should be sending a 400 here.  
> 400 Bad Request
 The server cannot or will not process the request due to an apparent client error (e.g., malformed request syntax, size too large, invalid request message framing, or deceptive request routing)

https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#4xx_Client_errors